web/src/main/java/org/cbioportal/web/MolecularProfileCorrelationController.java

```diff
@@ -0,0 +1,67 @@
+package org.cbioportal.web;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.cbioportal.model.MolecularProfileCorrelation;
+import org.cbioportal.service.MolecularProfileCorrelationService;
+import org.cbioportal.web.config.annotation.InternalApi;
+import org.cbioportal.web.parameter.CoExpressionFilter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.security.access.prepost.PreAuthorize;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@InternalApi
+@RestController
+@Validated
+@Api(tags = "Correlations", description = " ")
+public class MolecularProfileCorrelationController {
+    
+    @Autowired
+    private MolecularProfileCorrelationService molecularProfileCorrelationService;
+
+    @PreAuthorize("hasPermission(#molecularProfileId, 'MolecularProfile', 'read')")
+    @RequestMapping(value = "/molecular-profiles/correlations/{geneticEntityId}/{molecularProfileIdA}/{molecularProfileIdB}/fetch",
```

Comment:
We use the path variables only for `is-a` relationships, so all these 3 variables should be query parameters. And the name `geneticEntityId` is confusing because in the database it's the internal auto-increment id but in here we are using it as Entrez Gene Id or Geneset Id. So I suggest you create 2 separate query parameters, entrezGeneId and genesetId, and throw an exception if both of them are non-null.